### PR TITLE
SIMD bugfix: Store, inliner

### PR DESCRIPTION
**Pull request by arunetm**
1. Updating simd store semantics to return the value being stored.
2. Fixing an inliner bug while replacing dst for store operations.

### CODE REVIEW

**Comment by arunetm** [comment]
@abchatra @LouisLaf Please review.

File: lib/Backend/Inline.cpp [hunk 1]
```diff
@@ -5334,11 +5334,19 @@ Inline::Simd128FixLoadStoreInstr(Js::BuiltinFunction builtInId, IR::Instr * call
...
 
         eaInstr1 = linkOpnd->GetStackSym()->m_instrDef;
         value = eaInstr1->GetSrc1();
         linkOpnd = eaInstr1->GetSrc2();
 
+        //Load value to be stored to dst. Store returns the value being stored.
+        if (dst)
+        {
+            IR::Instr * ldInstr = IR::Instr::New(Js::OpCode::Ld_A, dst, value, callInstr->m_func);
+            callInstr->InsertAfter(ldInstr);
```
Comment (abchatra) [inline lib/Backend/Inline.cpp]: Nit: Can this be changed to InsertBefore and done at the end? InsertAfter is hard to read.

File: lib/Runtime/Language/AsmJsByteCodeGenerator.cpp [hunk 2]
```diff
@@ -1944,32 +1944,27 @@ namespace Js
...
+        if (simdFunction->IsSimdStoreFunc()) //Store
         {
+            // Arg3 - Value to Store. Builtin returns the value being stored.
             Assert(valueNode);
-            // Emit 3rd argument
-            valueInfo = Emit(valueNode);
-            if (valueInfo.type != simdFunction->GetArgType(2))
+            emitInfo = Emit(valueNode);
+
+            if (emitInfo.type != simdFunction->GetArgType(2))
```
Comment (abchatra) [inline lib/Runtime/Language/AsmJsByteCodeGenerator.cpp]: Can you please add the issue or VSO bug number corresponding to these changes?

Comment (nmostafa) [inline lib/Runtime/Language/AsmJsByteCodeGenerator.cpp]: There is no issue/bug related to this. We uncovered it while enabling store dst opnd.

Comment (abchatra) [inline lib/Runtime/Language/AsmJsByteCodeGenerator.cpp]: If this has to go for 1.2 then you need a bug or issue. Please file an issue with repro steps.

**Comment by nmostafa** [comment]
Fixes #708

File: lib/Backend/Inline.cpp (hunk 1 above)
Comment (arunetm) [inline lib/Backend/Inline.cpp]: Sure. Will use insert after. This change will be applied to the commit in the new PR for release 1.2

**Comment by arunetm** [comment]
Closing this one, and opening a new PR against release/1.2

**Pull request closed without merging**